src/features/DebugSession.ts

```diff
@@ -1,96 +1,96 @@
 
-enum DebugConfig {
+export enum DebugConfig {
     LaunchCurrentFile,
     LaunchScript,
     InteractiveSession,
     AttachHostProcess,
 }
 
+/** Make the implicit behavior of undefined and null in the debug api more explicit  */
+type PREVENT_DEBUG_START = undefined;
+type PREVENT_DEBUG_START_AND_OPEN_DEBUGCONFIG = null;
+type ResolveDebugConfigurationResult = DebugConfiguration | PREVENT_DEBUG_START | PREVENT_DEBUG_START_AND_OPEN_DEBUGCONFIG;
+
+const PREVENT_DEBUG_START = undefined;
+const PREVENT_DEBUG_START_AND_OPEN_DEBUGCONFIG = null;
+
+
+export const defaultDebugConfigurations: Record<DebugConfig, DebugConfiguration> = {
+    [DebugConfig.LaunchCurrentFile]: {
+        name: "PowerShell: Launch Current File",
+        type: "PowerShell",
+        request: "launch",
+        script: "${file}",
+        args: [],
+    },
+    [DebugConfig.LaunchScript]: {
+        name: "PowerShell: Launch Script",
+        type: "PowerShell",
+        request: "launch",
+        script: "Enter path or command to execute, for example: \"${workspaceFolder}/src/foo.ps1\" or \"Invoke-Pester\"",
+        args: [],
+    },
+    [DebugConfig.InteractiveSession]: {
+        name: "PowerShell: Interactive Session",
+        type: "PowerShell",
+        request: "launch",
+    },
+    [DebugConfig.AttachHostProcess]: {
+        name: "PowerShell: Attach to PowerShell Host Process",
+        type: "PowerShell",
+        request: "attach",
+        runspaceId: 1,
+    }
+};
+
 export class DebugSessionFeature extends LanguageClientConsumer
-    implements DebugConfigurationProvider, vscode.DebugAdapterDescriptorFactory {
+    implements DebugConfigurationProvider, DebugAdapterDescriptorFactory {
 
     private sessionCount = 1;
     private tempDebugProcess: PowerShellProcess | undefined;
     private tempSessionDetails: IEditorServicesSessionDetails | undefined;
     private handlers: vscode.Disposable[] = [];
-    private configs: Record<DebugConfig, DebugConfiguration> = {
-        [DebugConfig.LaunchCurrentFile]: {
-            name: "PowerShell: Launch Current File",
-            type: "PowerShell",
-            request: "launch",
-            script: "${file}",
-            args: [],
-        },
-        [DebugConfig.LaunchScript]: {
-            name: "PowerShell: Launch Script",
-            type: "PowerShell",
-            request: "launch",
-            script: "Enter path or command to execute, for example: \"${workspaceFolder}/src/foo.ps1\" or \"Invoke-Pester\"",
-            args: [],
-        },
-        [DebugConfig.InteractiveSession]: {
-            name: "PowerShell: Interactive Session",
-            type: "PowerShell",
-            request: "launch",
-        },
-        [DebugConfig.AttachHostProcess]: {
-            name: "PowerShell: Attach to PowerShell Host Process",
-            type: "PowerShell",
-            request: "attach",
-            runspaceId: 1,
-        },
-    };
+    private configs = defaultDebugConfigurations;
```

Comment:
Is this field still used? Could `defaultDebugConfigurations` be used instead? I didn't see it get modified later but maybe I missed it.

Reply:
It isn't used internally there, I just didn't catch it because it was referenced in tests. Fixed aa8a1ade40ba546671284d54d94b9a65de279303